WordPressAuthenticator/Signin/LoginSiteAddressViewController.swift

```diff
@@ -194,7 +194,7 @@ class LoginSiteAddressViewController: LoginViewController, NUXKeyboardResponder
         let baseSiteUrl = WordPressAuthenticator.baseSiteURL(string: loginFields.siteAddress)
         let successBlock: (WordPressComSiteInfo) -> Void = { [weak self] siteInfo in
             self?.loginFields.meta.siteInfo = siteInfo
-            if WordPressAuthenticator.shared.delegate?.allowWPComLogin == false {
+            if WordPressAuthenticator.shared.delegate?.allowWPComLogin == false && !siteInfo.hasJetpack {
```

Comment:
So... there are a few things going on in `fetchSiteInfo` and its associated code paths that seem off to me. For example, the `else` block in the conditional following the success block is never executed. I suspect there might be more work to be done in this method.  That said....

I think this change might introduce a hidden bug -- one that is not immediately obvious nor experienced but _could_ be if something else changed.  The logic here says: "If the delegate does not allow a wpcom login and the site does not have Jetpack, then show the prompt".  This works for WordPress.com sites since they do not have Jetpack since the conditional evalutes to false.  This works for self-hosted sites that have Jetpack since the conditional evaluates to true.  This _would fail_ for self-hosted sites without Jetpack if it was ever evaluated for one.  Currently the code path for a self-hosted non-Jetpack site does not encounter this check since the API call results in the failure block being executed.   Subtle, eh? 

I wonder if a more robust fix would be to just call the unauthenticated endpoint.  The `connect/site-info` results include a filed for `isWordPressDotCom` which could be checked first.  Then if true the delegate could be consulted about whether to allow a dotcom login.  Its a larger change but maybe worth it to clarify the work being done (and potentially clean up the other code paths that go through here.)

Thoughts?

Reply:
>This would fail for self-hosted sites without Jetpack if it was ever evaluated for one
Oh! I didn't realize that.

I'm going to give your proposed solution a try and see if it makes more sense. I love the idea of cleaning this method up a bit.